projects/FW103/src/main.c

```diff
@@ -0,0 +1,52 @@
+#include <stdint.h>
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+GpioAddress reading_addr = { .port = GPIO_PORT_A, .pin = 6 };
+GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+GpioSettings reading_settings = { .direction = GPIO_DIR_IN,
+                                  .state = GPIO_STATE_LOW,
+                                  .alt_function = GPIO_ALTFN_NONE,
+                                  .resistor = GPIO_RES_NONE };
+
+GpioSettings button_settings = { .direction = GPIO_DIR_IN,
+                                 .state = GPIO_STATE_LOW,
+                                 .alt_function = GPIO_ALTFN_NONE,
+                                 .resistor = GPIO_RES_NONE };
+
+InterruptSettings interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
```

Comment:
These are static variables (file local), so according to the coding standards they should be `static` and prefixed with `s_`. However, I don't think any of these need to be global, so please declare them inside a function instead.

Global/static variables should generally be minimized for several reasons, one of which is that the microcontrollers we program on have only ~16KB of memory. Global variables take up some of that memory all the time, whereas local variables only take it up when the relevant function is running, so local variables are preferred.